src/LookupArrays/selector.jl

```diff
@@ -103,16 +103,30 @@ function at(lookup::NoLookup, sel::At; kw...)
     return r
 end
 function at(lookup::LookupArray, sel::At; kw...)
-    at(order(lookup), lookup, val(sel), atol(sel), rtol(sel); kw...)
+    at(order(lookup), span(lookup), lookup, val(sel), atol(sel), rtol(sel); kw...)
 end
 function at(
-    ::Ordered, lookup::LookupArray{<:Union{Number,Dates.TimeType}}, selval, atol, rtol::Nothing;
+    ::Ordered, span::Regular, lookup::LookupArray{<:Integer}, selval, atol::Nothing, rtol::Nothing;
+    err=_True()
+)
+    x = unwrap(selval)
+    Δ = step(span)
+    i, remainder = divrem(x - first(lookup), Δ)
+    i += 1
+    if remainder == 0 && checkbounds(Bool, lookup, i)
+        return i
```

Comment:
Is this still correct for an `OffsetArray`? It would be nice to test for this in the ecosystem.jl tests.

I imagine you will need `i += firstindex(lookup)` rather than `i += 1`, which seems to imbed a `fistindex(lookup) == 1` assumption, unless I misunderstand something.